backend/data/src/main/kotlin/io/tolgee/formats/apple/in/xcstrings/XcstringsFileProcessor.kt

```diff
@@ -41,33 +44,77 @@ class XcstringsFileProcessor(
       context.addKeyDescription(key, comment)
     }
 
+    if (!localizations.has(sourceLanguage)) {
+      val converted = messageConvertor.convert(
+        rawData = key,
+        languageTag = sourceLanguage,
+        convertPlaceholders = context.importSettings.convertPlaceholdersToIcu,
+        isProjectIcuEnabled = context.projectIcuPlaceholdersEnabled
+      )
+
+      context.addTranslation(
+        keyName = key,
+        languageName = sourceLanguage,
+        value = converted.message,
+        convertedBy = importFormat,
+        state = TranslationState.TRANSLATED,
+        rawData = key,
+        pluralArgName = converted.pluralArgName
+      )
+    }
+
     localizations.fields().forEach { (languageTag, localization) ->
       when {
         localization.has("stringUnit") -> {
           processSingleTranslation(key, languageTag, localization)
         }
+
         localization.has("variations") -> {
           processPluralTranslation(key, languageTag, localization)
         }
       }
     }
   }
 
+
+  private fun convertXCtoTolgeeTranslationState(xcState: String?): TranslationState {
+    return when (xcState) {
+      "needs_review" -> TranslationState.TRANSLATED
+      "translated" -> TranslationState.REVIEWED
+      null -> TranslationState.UNTRANSLATED
+      else -> TranslationState.UNTRANSLATED
+    }
+  }
+
   private fun processSingleTranslation(
     key: String,
     languageTag: String,
     localization: JsonNode,
   ) {
     val stringUnit = localization.get("stringUnit")
+    val xcState = stringUnit?.get("state")?.asText()
+    var translationState = convertXCtoTolgeeTranslationState(xcState)
+
     val translationValue = stringUnit?.get("value")?.asText()
 
     if (translationValue != null) {
+      val converted = messageConvertor.convert(
```

Comment:
It seems like the conversion and adding translation logic can be extracted to separate function for this case and also for the case when the sourceLanguage is missing.